react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentImagePosition: 0,
+      setIntervalStatus: null
+    };
+    this.handleArrowClick = this.handleArrowClick.bind(this);
+    this.handleCircleClick = this.handleCircleClick.bind(this);
+  }
+
+  componentDidMount() {
+    this.setState({
+      setIntervalStatus: setInterval(() => {
+        if (this.state.currentImagePosition >= this.props.content.length - 1) {
+          this.setState({ currentImagePosition: 0 });
+        } else {
+          this.setState({ currentImagePosition: this.state.currentImagePosition + 1 });
+        }
+      }, 3000)
+    });
+  }
+
+  handleArrowClick(event) {
+    clearInterval(this.state.setIntervalStatus);
+    this.componentDidMount();
+    if (event.target.className === 'fas fa-chevron-circle-right fa-2x') {
+      this.setState({ currentImagePosition: this.state.currentImagePosition + 1 });
+      if (this.state.currentImagePosition >= this.props.content.length - 1) {
+        this.setState({ currentImagePosition: 0 });
+      }
+    }
+    if (event.target.className === 'fas fa-chevron-circle-left fa-2x') {
+      this.setState({ currentImagePosition: this.state.currentImagePosition - 1 });
+      if (this.state.currentImagePosition <= 0) {
+        this.setState({ currentImagePosition: this.props.content.length - 1 });
+      }
+    }
+  }
+
+  handleCircleClick(event) {
+    clearInterval(this.state.setIntervalStatus);
+    this.componentDidMount();
+    const targetedCircle = Number(event.target.id);
+    if (event.target.className === 'far fa-circle ml-10') {
+      this.setState({ currentImagePosition: targetedCircle });
+    }
```

Comment:
In general we avoid calling lifecycle methods in our own code and let react call the. But, you _can_ define a separate method like `autoplay()` and call that method inside of `componentDidMount()` and `handleCircleClick()`